From 5e15d8b2ffd632534e4656a7632e6b672024d0f0 Mon Sep 17 00:00:00 2001
From: David Eads <deads@redhat.com>
Date: Fri, 9 Jul 2021 10:30:56 -0400
Subject: [PATCH] UPSTREAM: 103612: tolerate additional, but congruent, events
 for integration test

OpenShift-Rebase-Source: 2f4c829d138

UPSTREAM: 103612: tolerate additional, but congruent, events for integration test
---
 test/integration/events/events_test.go | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/test/integration/events/events_test.go b/test/integration/events/events_test.go
index 2df67fd8b44..a89426b49cd 100644
--- a/test/integration/events/events_test.go
+++ b/test/integration/events/events_test.go
@@ -79,7 +79,9 @@ func TestEventCompatibility(t *testing.T) {
 			return false, err
 		}
 
-		if len(v1Events.Items) != 2 {
+		// Be sure that at least the events we sent in the test were delivered.
+		// To add any events from the kube-apiserver itself will require this tolerate additional events.
+		if len(v1Events.Items) < 2 {
 			return false, nil
 		}
 
@@ -88,9 +90,15 @@ func TestEventCompatibility(t *testing.T) {
 			return false, err
 		}
 
-		if len(events.Items) != 2 {
+		if len(events.Items) < 2 {
 			return false, nil
 		}
+
+		// Be sure that both APIs produce the same number of events.
+		if len(events.Items) != len(v1Events.Items) {
+			return false, nil
+		}
+
 		return true, nil
 	})
 	if err != nil {
@@ -137,7 +145,9 @@ func TestEventSeries(t *testing.T) {
 			return false, err
 		}
 
-		if len(events.Items) != 1 {
+		// Be sure that at least the events we sent in the test were delivered.
+		// To add any events from the kube-apiserver itself will require this tolerate additional events.
+		if len(events.Items) < 1 {
 			return false, nil
 		}
 
@@ -145,7 +155,7 @@ func TestEventSeries(t *testing.T) {
 			return false, nil
 		}
 
-		if events.Items[0].Series.Count != 2 {
+		if events.Items[0].Series.Count < 2 {
 			return false, fmt.Errorf("expected EventSeries to have a starting count of 2, got: %d", events.Items[0].Series.Count)
 		}
 
-- 
2.46.0

